### DISCUSSION TITLE
Write configuration in code (like XML)?

### DISCUSSION THREAD

**Discussion by axiskahoy** [bitfocus/companion#2998] [Q&A]
Hi everyone! Sorry if this question isn't new or is obvious, but is there a way to write the deck configurations in code ? In the same way I saw some of my colleagues write some ATEM configurations directly in XML. I tried to open a config page file in Visual Studio but it can't read it. Note I am a beginner both in companion and coding, and an AV technician apprentice. Thanks !

**Comment by Julusian** [comment]
It is possible to, the exports are json (sometimes gzipped, which is optional) so can be edited. But it is generally advised not to, as we don't validate the imports much, so if you do something wrong, it could 'corrupt' your companion config.

And the structure of this json has not been designed to be friendly to edit, so it has some odd quirks, it is simply snippets of the internal database

This is related to https://github.com/bitfocus/companion/issues/1339

**Reply by axiskahoy** [reply]
Thanks for your answer ! How do you export the code ? For now I'm just playing with the app so I'm not afraid to lose content
